### PR TITLE
Interactive Radio and Window in Birdie room

### PR DESCRIPTION
**Pull request by masonachu**
2020-03-15 02_59_55-EspionAge - MAIN - PC, Mac   Linux Standalone - Unity 2019 3 0f6 Personal  PREVI

additional script edits were made on:
- DoorBehaviour (sfx on door open and slam setup but not yet implemented)

### CODE REVIEW

File: Assets/Scripts/DoorBehaviour.cs [hunk 1]
```diff
@@ -36,17 +42,21 @@ private void Update()
...
             }
-        } 
+        }
         else
         {
             if (isClosed)
             {
                 OnDoorOpen?.Invoke();
+                FMODUnity.RuntimeManager.PlayOneShot(DoorSqueak, transform.position);
+                Debug.Log("Played FMOD");
```
Comment (jurcaua) [inline Assets/Scripts/DoorBehaviour.cs:59]: I'd remove if you don't need anymore.

File: Assets/Scripts/Audio/FMODMissionComplete.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    private bool completed;
+    private bool SeekSpeed = true;
+
+    void OnTriggerEnter(Collider other)
+    {
+        if (other.tag == "Player" == !completed)
+        {
+            FMODUnity.RuntimeManager.StudioSystem.setParameterByName(param, 1f, SeekSpeed);
+            completed = true;
+            Debug.Log("param 1f");
```
Comment (jurcaua) [inline Assets/Scripts/Audio/FMODMissionComplete.cs]: I'd remove this and the other Debug.Log statement below if you don't need it anymore.

**Review by jurcaua — COMMENTED** [review]
Just small comments about removing some Debug.Log calls if we dont need them!
Gonna wait for ur change we talked about for FMODMissionComplete 😄 
amaze stuff though, so many assets, you're our savior :godmode:

File: Assets/Scripts/Audio/FMODRadio.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        {
+            musicEv.stop(FMOD.Studio.STOP_MODE.IMMEDIATE);
+            isPlaying = false;
+        }
+
+        if (!isPlaying)
+        {
+            musicEv.start();
+            isPlaying = true;
+        }
```
Comment (chadleytan) [inline Assets/Scripts/Audio/FMODRadio.cs:36]: I just realized we can make this much simpler if you just want to change the station
```suggestion
        if (isPlaying)
        {
            musicEv.stop(FMOD.Studio.STOP_MODE.IMMEDIATE);
            musicEv.start();
        }
```

Comment (masonachu) [reply Assets/Scripts/Audio/FMODRadio.cs:36]: made the edit and works perfectly!

File: Assets/Scripts/DoorBehaviour.cs (hunk 1 above)
Comment (masonachu) [reply Assets/Scripts/DoorBehaviour.cs:59]: Done!

File: Assets/Scripts/Audio/FMODMissionComplete.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    void OnTriggerEnter(Collider other)
+    {
+        if (!isComplete)
+        {
+            FMODUnity.RuntimeManager.StudioSystem.setParameterByName(parameter, 1f);
+            isComplete = true;
+        }
+    }
+
+    private void missionComplete(Interactable source)
```
Comment (jurcaua) [inline Assets/Scripts/Audio/FMODMissionComplete.cs]: Usual function naming convention for functions is starting with a capital letter, also I'd recommend maybe the name `OnMissionComplete` instead?

**Review by jurcaua — CHANGES_REQUESTED** [review]
Small small change, but otherwise looks amazing! 10/10 my man

File: Assets/Scripts/Audio/FMODMissionComplete.cs (hunk 4 above)
Comment (masonachu) [reply Assets/Scripts/Audio/FMODMissionComplete.cs]: Done!

**Review by jurcaua — APPROVED** [review]
💋 niceu worku mr masonu-san

**Pull request merged**